src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/common/config/PersistenceHelper.kt

```diff
@@ -0,0 +1,170 @@
+package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.common.config
+
+import jakarta.persistence.EntityManager
+import jakarta.persistence.PersistenceContext
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDateTime
+import java.util.UUID
+
+@Component
+@Transactional
+class PersistenceHelper {
+
+  @PersistenceContext
+  private lateinit var entityManager: EntityManager
+
+  fun clearAllTableContent() {
+    entityManager.createNativeQuery("DELETE FROM course_participation").executeUpdate()
+    entityManager.createNativeQuery("DELETE FROM referral").executeUpdate()
+    entityManager.createNativeQuery("DELETE FROM offering").executeUpdate()
+    entityManager.createNativeQuery("DELETE FROM course_audience").executeUpdate()
+    entityManager.createNativeQuery("DELETE FROM course").executeUpdate()
+    entityManager.createNativeQuery("DELETE FROM audience").executeUpdate()
+    entityManager.createNativeQuery("DELETE FROM referrer_user").executeUpdate()
+  }
+
+  fun setupCourses() {
```

Comment:
Rather than creating everything up front and blowing it away between tests, how would you feel about extracting these into some helper like `createCourse(id:)` that inserts the specific entities with the ids we want, that could then be called in each `@State`? That way it would be really easy to tell exactly what entities we have lying about in each test state, and we wouldn't have to create so many things for each test?

I don't think this needs to block the PR, as this is definitely a fantastic step in the right direction, but having a smaller number of entities in each state will definitely help readability etc. This could be an enhancement of the current approach.